packages/fast-usdc/src/util/cctp.js

```diff
@@ -0,0 +1,65 @@
+import { Buffer } from 'node:buffer';
+import { bech32 } from 'bech32';
+import { ethers } from 'ethers';
+
+/**
+ * Adapted from https://docs.noble.xyz/cctp/mint#encoding
+ *
+ * @param {string} address
+ * @returns {string}
+ */
+export const encodeBech32Address = address => {
+  const decoded = bech32.decode(address);
+  const rawBytes = Buffer.from(bech32.fromWords(decoded.words));
+
+  const padded = Buffer.alloc(32);
+  rawBytes.copy(padded, 32 - rawBytes.length);
+
+  return `0x${padded.toString('hex')}`;
+};
+
+const tokenAbi = ['function approve(address spender, uint256 value) external'];
+
+const contractAbi = [
+  'function depositForBurn(uint256 amount, uint32 destinationDomain, bytes32 mintRecipient, address burnToken) external',
+];
+
+export const depositForBurn = async (
+  /** @type {string} */ ethRpc,
```

Comment:
How about passing in a provider instead of a string?

Just like passing a file object instead of a filename.

Reply:
This should make unit testing transfer straightforward

Reply:
Done